src/diffusers/image_processor.py

```diff
@@ -168,6 +168,20 @@ def postprocess(
         image,
         output_type: str = "pil",
     ):
+        if output_type not in ["latent", "pt", "np", "pil"]:
+            deprecation_message = (
+                f"the output_type {output_type} is outdated and has been set to `np`. Please make sure to set it to one of these instead: "
+                "`pil`, `np`, `pt`, `latent`"
+            )
+            deprecate("Unsupported output_type", "1.0.0", deprecation_message, standard_warn=False)
+            output_type = "np"
+
+        if output_type == "latent":
+            return image
+
```

Comment:
```suggestion

if not isinstance(do_normalize, list):
    do_normalize = image.shape[0] * [do_normalize or self.config.do_normalize]
```

Reply:
I think we will need this to support the blacked-out image

Reply:
@patrickvonplaten 

can we just check from the `postprocess` to see if the image is black? And not denormalize if so?

I will do what you proposed here but I just want to understand and think it will help me to make better design decisions in the future.

Reply:
We could do this as well, I also thought about this. I'm a bit worried though that it's a bit black magic. In the 1 in a million cases where SD produces an image that has exactly only 0s, we should actually normalize it to an image that has only 0.5. It's highly unlikely, but it could happen. So in this sense there is a difference between a "blacked-out" image due to NSFW and a "blacked-out" image due to SD

In a sense the postprocessor should not have to know anything about the safety checker - they should be as orthogonal / disentangled as possible. But I definitely understand your point here, it's quite some extra code.

Reply:
I don't like the black magic that checks whether the image is black. I think the chance for SD to produce a full black image is really remote, but it's strange that the post-processor has to be aware about safety checker images.

Another alternative would be for `run_safety_checker` to _normalize_ black images so it returns `-1`s onstead of `0`s. This is consistent with the fact that `run_safety_checker` receives normalized images and also returns normalized images **if they are ok**, but it returns denormalized images (black represented as `0`) if they are NSFW.

We could do something like this after [this line](https://github.com/huggingface/diffusers/blob/7c1bb9ae81c97f6f10cf36784c17a51ebddeea08/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L443-L445):

```py
            image = torch.stack([self.image_processor.normalize(image[i]) if has_nsfw_concept[i] else image[i] for i in range(image.shape[0])])
```

And then we could remove the `do_normalize` argument here.

Would that be preferrable?

Reply:
ohh I think it makes sense to return `-1` for NSFW pixels then `post_processor` doesn't have to treat it differently
but we also need to make sure `run_safety_checker` is backward-compatible

maybe we can do this only when the image is a PyTorch tensor? I think that would be a little bit confusing no? 
```python
if torch.is_tensor(image):            
    image = torch.stack([self.image_processor.normalize(image[i]) if has_nsfw_concept[i] else image[i] for i in range(image.shape[0])])
```